### PR TITLE
feat(mysql): #277 STRICT_ALL_TABLES required sql mode support

### PR DESCRIPTION
**Pull request by deeptibaghel**
Fixes https://github.com/mozilla/fxa-auth-db-mysql/issues/277

- Added the query to set the sql_mode
- Added unit test to disallow any other sql_mode

### CODE REVIEW

File: lib/db/mysql.js [hunk 1]
```diff
@@ -1138,14 +1155,20 @@ module.exports = function (log, error) {
           return resolve(connection)
         }
 
-        connection.query('SET NAMES utf8mb4 COLLATE utf8mb4_bin;', (err) => {
+        var mode = REQUIRED_SQL_MODES.join(',')
+        connection.query('SET SESSION sql_mode = \'' + mode + '\'', function(err) {
```
Comment (vladikoff) [inline lib/db/mysql.js]: can this use \` \` for the query here? might make it cleaner https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

**Comment by deeptibaghel** [comment]
@rfk made the changes, will squash commits after review :)

File: lib/db/mysql.js [hunk 2]
```diff
@@ -1156,7 +1156,7 @@ module.exports = function (log, error) {
         }
 
         var mode = REQUIRED_SQL_MODES.join(',')
-        connection.query('SET SESSION sql_mode = \'' + mode + '\'', function(err) {
+        connection.query(`SET SESSION sql_mode = ${mode};`, (err) => {
```
Comment (vladikoff) [inline lib/db/mysql.js]: you *might* need the `' '` around ${mode} per example in https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html

`SET SESSION sql_mode = 'modes';`

Comment (deeptibaghel) [reply lib/db/mysql.js]: Yeah i changed it after this, need to squash commits :(

File: test/local/mysql_tests.js [hunk 3]
```diff
@@ -41,6 +41,26 @@ describe('MySQL', () => {
...
+        'STRICT_ALL_TABLES',
+        'NO_ENGINE_SUBSTITUTION',
+      ]
+      var mode = REQUIRED_SQL_MODES.join(',')
+      configSqlMode.sql_mode = 'xyz'
+      return DB.connect(configSqlMode)
+        .then(
+          assert.fail,
+          err => {
+            assert.equal(err.message, `You cannot use any sql mode other than ${mode}`)
```
Comment (rfk) [inline test/local/mysql_tests.js]: 👍 great to have this test to make sure!

**Comment by deeptibaghel** [comment]
Thanks @rfk , have squashed the commits :)

**Comment by vladikoff** [comment]
Thank you @deeptibaghel !!

**Pull request merged**